File: MultipleItem.js
const Item = require("./Item");

class MultipleItem extends Item {
  constructor(name, unity) {
    super(name);
    this.quantity = 0;
    this.unity = unity;
  }

  add(quantityToAdd) {
    this.quantity += quantityToAdd;
    return this;
  }

  take(quantityToTake) {
    if (this.quantity > quantityToTake) {
      this.quantity -= quantityToTake;
      return this;
    } else {
      console.log(
        `${this.getName()} does not have sufficient quantity for this transaction.`,
      );
    }
    return null;
  }

  getQuantity() {
    return this.quantity;
  }

  getQuantityWithUnity() {
    return `${this.quantity} ${this.unity}`;
  }

  setQuantity(quantity) {
    this.quantity = quantity;
  }

  setUnity(unity) {
    this.unity = unity;
  }

  getUnity() {
    return this.unity;
  }
}

module.exports = MultipleItem;

File: MultipleItem.py
from Item import Item


class MultipleItem(Item):
    def __init__(self, new_name, unity):
        super().__init__(new_name)
        self.unity = unity
        self.quantity = 0

    def add(self, quantity_to_add):
        self.quantity += quantity_to_add
        return self

    def take(self, quantity_to_take):
        if self.quantity > quantity_to_take:
            self.quantity -= quantity_to_take
            return self
        else:
            print(f"{super().name} does not have sufficient quantity for this transaction.")
            return None

    def get_quantity(self):
        return self.quantity

    def get_quantity_with_unity(self):
        return f"{self.quantity} {self.unity}"

    def set_quantity(self, quantity):
        self.quantity = quantity

    def set_unity(self, unity):
        self.unity = unity

    def get_unity(self):
        return self.unity
